python_modules/libraries/dagster-dg/dagster_dg/mcp/server.py

```diff
@@ -1,67 +1,229 @@
+
+    Args:
+        project_path: The full path to your Dagster project.
+
+    Returns:
+        The list of components available for the given Dagster project.
+    """
+    return _subprocess(
+        [
+            "uv",
+            "run",
+            "dg",
+            "list",
+            "plugins",
+            "--feature",
+            "component",
+            "--json",
+        ],
         cwd=project_path,
-    ).decode("utf-8")
+    )
 
 
 @mcp.tool()
-async def list_dagster_components(project_path: str) -> str:
-    """List all Dagster components for the project. Components
-    can be scaffolded to create new Dagster definitions. Call this
-    when you need to add new Dagster definitions and want to get a list
-    of components to use to do so.
+async def install_component(project_path: str, package_name: str) -> str:
+    """Install a component from the marketplace.
+
+    Some available components include:
+        - `dagster-dbt`
+        - `dagster-sling`
+        - `dagster-evidence`
+
+    Args:
+        project_path: The full path to your Dagster project.
+        package_name: The Python package available on pypi for the component.
+
+    Returns:
+        The output of the `uv add` command
     """
-    return subprocess.check_output(
-        ["uv", "run", "dg", "list", "plugins", "--feature", "component"],
+    # TODO - get list of available components from the registry
+    return _subprocess(
+        ["uv", "add", package_name],
         cwd=project_path,
-    ).decode("utf-8")
+    )
 
 
 @mcp.tool()
 async def scaffold_dagster_component(
-    project_path: str, component_type: str, component_name: str
+    project_path: str,
+    component_type: str,
+    component_name: str,
 ) -> str:
-    """Scaffold a new Dagster component in the project. This produces
-    a component.yaml file which can be populated with the appropriate
-    fields for the component.
+    """Scaffold a new Dagster component in the project.
+
+    This produces a component.yaml file which can be populated with the appropriate fields for the component.
+
+    Args:
+        project_path: The full path to your Dagster project.
+        component_type: The full identifier of the component to be scaffolded (e.g. dagster_sling.SlingReplicationCollectionComponent)
+        component_name: The name of the component to provide for the newly scaffolded component.
+
+    Returns:
+        The output from running the `dg scaffold` command.
     """
-    return subprocess.check_output(
-        ["uv", "run", "dg", "scaffold", component_type, component_name],
+    # NOTE: attempted to use `*args` and optional parameter without success
+
+    return _subprocess(
+        ["uv", "run", "dg", "--verbose", "scaffold", component_type, component_name],
         cwd=project_path,
-    ).decode("utf-8")
+    )
+
+
+# TODO - remove once *args is supported in scaffold component
+@mcp.tool()
+async def scaffold_dagster_dbt_component(
```

Comment:
unfortunate we need this right now but seems fine as a temp measure

Reply:
Surprisingly it looks like I was able to get it to work by adding a `--help` tool.
<img width="813" alt="image" src="https://github.com/user-attachments/assets/7fa4b1ee-868f-4f7a-abca-a449d4b50110" />